monGARS/core/distributed_scheduler.py

```
@@ -25,44 +116,217 @@ def __init__(
+        )
+        try:
+            while self._running or not self.queue.empty():
+                try:
+                    factory = await asyncio.wait_for(self.queue.get(), timeout=1)
+                except asyncio.TimeoutError:
+                    if not self._running and self.queue.empty():
+                        break
+                    continue
+                try:
+                    result = await factory()
+                except Exception as exc:  # pragma: no cover - unexpected errors
```

Ruff (0.13.1)

Local variable `exc` is assigned to but never used

Remove assignment to unused variable `exc`

(F841)